flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/ScopeSnapshot.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.calcite.sql.validate;
+
+import org.apache.calcite.schema.SchemaVersion;
+import org.apache.calcite.sql.SqlNode;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.List;
+
+/** Represent the snapshot of the {@link SqlValidatorScope} */
+public class ScopeSnapshot extends DelegatingScope {
+    private SqlValidatorSnapshot sqlValidatorSnapshot;
+
+    public ScopeSnapshot(SqlValidatorScope parent, SchemaVersion schemaVersion) {
+        super(parent);
+        this.sqlValidatorSnapshot =
+                new SqlValidatorSnapshot((SqlValidatorImpl) parent.getValidator(), schemaVersion);
+    }
+
+    @Override
+    public void resolveTable(
+            List<String> names, SqlNameMatcher nameMatcher, Path path, Resolved resolved) {
+        new EmptyScope(sqlValidatorSnapshot).resolveTable(names, nameMatcher, path, resolved);
```

Comment:
Please add a comment to explain why we can use `EmptyScope` directly. 
IIUC, the reason is. in the time travel case, the parent of the ` ScopeSnapshot` will always be `CatalogScope` with an `EmptyScope` as the parent of the `CatalogScope` ?